tools/test/h5dump/CMakeTests.cmake

```diff
@@ -454,462 +454,256 @@ macro (ADD_HELP_TEST testname resultcode)
-    endif ()
-    add_test (
-        NAME H5DUMP-${resultfile}-output-cmp
-        COMMAND ${CMAKE_COMMAND} -E compare_files --ignore-eol ${resultfile}.txt ${resultfile}.exp
-    )
-    set_tests_properties (H5DUMP-${resultfile}-output-cmp PROPERTIES
-        DEPENDS H5DUMP-${resultfile}
-        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/testfiles/std"
-    )
-    add_test (
-        NAME H5DUMP-${resultfile}-output-cmp-ddl
-        COMMAND ${CMAKE_COMMAND} -E compare_files --ignore-eol ${ddlfile}.txt ${ddlfile}.exp
-    )
-    set_tests_properties (H5DUMP-${resultfile}-output-cmp-ddl PROPERTIES
-        DEPENDS H5DUMP-${resultfile}-output-cmp
-        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/testfiles/std"
-    )
-    if ("H5DUMP-${resultfile}-output-cmp-ddl" MATCHES "${HDF5_DISABLE_TESTS_REGEX}")
-      set_tests_properties (H5DUMP-${resultfile}-output-cmp-ddl PROPERTIES DISABLED true)
-    endif ()
-  endif ()
-  add_test (
-      NAME H5DUMP-${resultfile}-clean-objects
-      COMMAND ${CMAKE_COMMAND} -E remove
-          ${ddlfile}.txt
-          ${resultfile}.txt
-  )
-  if (HDF5_ENABLE_USING_MEMCHECKER)
-    set_tests_properties (H5DUMP-${resultfile}-clean-objects PROPERTIES
-        DEPENDS H5DUMP-${resultfile}
-        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/testfiles/std"
-    )
+    set (BINARY_OUTPUT_FLAG "")
+  endif()
+
+  if (DEFINED ${ARG_RESULT_CHECK})
+    set (ARG_RESULT_CHECK_FILE "${ARG_RESULT_CHECK}")
   else ()
-    set_tests_properties (H5DUMP-${resultfile}-clean-objects PROPERTIES
-        DEPENDS H5DUMP-${resultfile}-output-cmp-ddl
-        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/testfiles/std"
-    )
+    set (ARG_RESULT_CHECK_FILE "${testname}.ddl")
   endif ()
-endmacro ()
 
-macro (ADD_H5_EXPORT_TEST resultfile targetfile resultcode)
-  if (NOT HDF5_ENABLE_USING_MEMCHECKER)
-    add_test (
-        NAME H5DUMP-output-${resultfile}-clear-objects
-        COMMAND ${CMAKE_COMMAND} -E remove
-            ${resultfile}.txt
-    )
-    set_tests_properties (H5DUMP-output-${resultfile}-clear-objects PROPERTIES
-        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/testfiles/std"
-    )
-    add_test (
-        NAME H5DUMP-output-${resultfile}
-        COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:h5dump> ${ARGN} ${resultfile}.txt ${targetfile}
-    )
-    set_tests_properties (H5DUMP-output-${resultfile} PROPERTIES
-        DEPENDS H5DUMP-output-${resultfile}-clear-objects
-        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/testfiles/std"
+  # Certain args are fully incompatible with memchecker; skip in these cases
+  set(should_skip_test FALSE)
+
+  if (HDF5_ENABLE_USING_MEMCHECKER)
+    if (DEFINED ARG_H5ERRREF OR ${ARG_BINFILE} OR ${ARG_GREP_COMPARE})
+      set(should_skip_test TRUE)
+    endif()
+  endif()
+
+  if (${ARG_SKIP_TEST})
+    set(should_skip_test TRUE)
+  endif()
+
+  if (should_skip_test)
+    # Add a test that just displays a skip message instead
+    add_test(
```

Comment:
Note that while the `ADD_SKIP_H5_TEST` macro wasn't actually used, what it was doing seems different than what we'd want now. Rather, a test should always be added (with its original name and its original command, not a command that just echoes "SKIP ..."), but when `SKIP_TEST` is specified in this new scheme,

```
set_tests_properties (H5DUMP-{ctest_testname} PROPERTIES DISABLED true)
```

should be called to specify that the test is disabled. That way the real test exists, but isn't run and CI can pick up that the test is being skipped when reporting results. It also keeps the test names consistent without having special test names for how a test is being run. This would also cleanup this particular part of the logic by flattening a level of `if ... else`.